testing/unittests/jlibtests.cpp

```diff
@@ -3658,47 +3664,40 @@ class PTreeSerializationDeserializationTest : public CppUnit::TestFixture
         originalTree.setown(tree);
         CCycleTimer timer;
 
-        // Serialization
         MemoryBuffer memoryBuffer, streamBuffer;
 
-        // Time serialize() method
         timer.reset();
         originalTree->serialize(memoryBuffer);
         __uint64 serializeElapsedNs = timer.elapsedNs();
         size_t memoryBufferSize = memoryBuffer.length();
 
-        // Time serializeToStream() method
         Owned<IBufferedSerialOutputStream> out = createBufferedSerialOutputStream(streamBuffer);
         timer.reset();
         originalTree->serializeToStream(*out);
         out->flush();
         __uint64 serializeToStreamElapsedNs = timer.elapsedNs();
         size_t streamBufferSize = streamBuffer.length();
 
-        // Validation - serialized data matches
         CPPUNIT_ASSERT_EQUAL(memoryBufferSize, streamBufferSize);
         CPPUNIT_ASSERT(memcmp(memoryBuffer.toByteArray(), streamBuffer.toByteArray(), memoryBufferSize) == 0);
 
-        // Time deserialize() method
-        Owned<IPropertyTree> memoryBufferDeserialized = createPTree();
+        // Copy streamBuffer for deserializationFromStream() tests
+        MemoryBuffer streamBuffer2, streamBuffer3;
+        streamBuffer2.append(streamBuffer.length(), streamBuffer.toByteArray());
+        streamBuffer3.append(streamBuffer.length(), streamBuffer.toByteArray());
+
         timer.reset();
-        memoryBufferDeserialized->deserialize(memoryBuffer);
+        Owned<IPropertyTree> memoryBufferDeserialized = createPTree(memoryBuffer);
         __uint64 deserializeElapsedNs = timer.elapsedNs();
 
-        // Time deserializeFromStream() method
         Owned<IBufferedSerialInputStream> in = createBufferedSerialInputStream(streamBuffer);
-        Owned<IPropertyTree> streamDeserialized = createPTree();
         timer.reset();
-        streamDeserialized->deserializeFromStream(*in);
+        Owned<IPropertyTree> streamDeserialized = createPTreeFromBinary(*in, ipt_none);
         __uint64 deserializeFromStreamElapsedNs = timer.elapsedNs();
 
-        // Create PTree from Binary tests
-        //
-        // Test 1: Call with null nodeCreator (should fall back to createPTree(src, ipt_none))
-        streamBuffer.reset();
-        Owned<IBufferedSerialInputStream> in2 = createBufferedSerialInputStream(streamBuffer);
+        Owned<IBufferedSerialInputStream> in2 = createBufferedSerialInputStream(streamBuffer2);
         Owned<IPropertyTree> deserializedCreatePTreeFromBinaryWithNull = createPTreeFromBinary(*in2, nullptr);
```

Comment:
nullptr is at odds with prototype, and therefore what does it mean .."WithNull"

Reply:
what's the difference between this and streamDeserialized? 1st was based on createBufferedSerialInputStream from streamBuffer, 2nd is based on createBufferedSerialInputStream of streamBuffer2 afaics - duplicatation/no clear if different/how different.